internal/liquids/nova/liquid.go

```diff
@@ -0,0 +1,112 @@
+/*******************************************************************************
+*
+* Copyright 2024 SAP SE
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You should have received a copy of the License along with this
+* program. If not, you may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*******************************************************************************/
+
+package nova
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/gophercloud/gophercloud/v2"
+	"github.com/gophercloud/gophercloud/v2/openstack"
+	"github.com/gophercloud/gophercloud/v2/openstack/compute/v2/flavors"
+	"github.com/sapcc/go-api-declarations/liquid"
+)
+
+type Logic struct {
+	NovaV2 *gophercloud.ServiceClient `yaml:"-"`
+}
+
+// Init implements the liquidapi.Logic interface.
+func (l *Logic) Init(ctx context.Context, provider *gophercloud.ProviderClient, eo gophercloud.EndpointOpts) (err error) {
+	l.NovaV2, err = openstack.NewComputeV2(provider, eo)
+	if err != nil {
+		return err
+	}
+	l.NovaV2.Microversion = "2.61" // to include extra specs in flavors.ListDetail()
+
+	return nil
+}
+
+// BuildServiceInfo implements the liquidapi.Logic interface.
+func (l *Logic) BuildServiceInfo(ctx context.Context) (liquid.ServiceInfo, error) {
+	resources := map[liquid.ResourceName]liquid.ResourceInfo{
+		"cores": {
+			Unit:        liquid.UnitNone,
+			HasCapacity: true,
+			HasQuota:    true,
+		},
+		"instances": {
+			Unit:        liquid.UnitNone,
+			HasCapacity: true,
+			HasQuota:    true,
+		},
+		"ram": {
+			Unit:        liquid.UnitMebibytes,
+			HasCapacity: true,
+			HasQuota:    true,
+		},
+		"server_groups": {
+			Unit:     liquid.UnitNone,
+			HasQuota: true,
+		},
+		"server_group_members": {
+			Unit:     liquid.UnitNone,
+			HasQuota: true,
+		},
+	}
+
+	err := FlavorSelection{}.ForeachFlavor(ctx, l.NovaV2, func(f flavors.Flavor) error {
+		if f.ExtraSpecs["capabilities:hypervisor_type"] == "ironic" {
+			return nil
+		}
+		if f.ExtraSpecs["quota:separate"] == "true" {
+			resources[liquid.ResourceName(f.Name)] = liquid.ResourceInfo{
```

Comment:
```suggestion
			resources[liquid.ResourceName("instances_"+f.Name)] = liquid.ResourceInfo{
```
Also, this is a candidate for a helper function, if this conversion from flavor name to resource name is needed more than once.